spring-integration-file/src/main/java/org/springframework/integration/file/FileReadingMessageSource.java

```diff
@@ -350,8 +350,21 @@ protected void onInit() {
 
 	@Override
 	public Message<File> receive() throws MessagingException {
+		AbstractIntegrationMessageBuilder<File> messageBuilder = doReceive();
+
 		Message<File> message = null;
 
+		if (messageBuilder != null) {
+			message = messageBuilder.build();
```

Comment:
Why are we building here? Just for the log message?

I guess we could move that log up to the abstract class and have a boolean flag.

Reply:
Because this is `FileReadingMessageSource` - clean `MessageSource` implementation which `receive()` method contract requires the `Message` to be returned.

Not sure what I am missing here...

Reply:
Hmmm - since I was just looking at the changes, I assumed this was an AMS - since the title of this PR is "Optimize AbstractMessageSources". But now I see it is not.

So, why is this refactoring in the PR at all, without a clarifying mention in the commit comment?

Reply:
I repushed with an updated commit message.
Was in a hurry in the evening to PR'ing this.